Card/src/main/java/com/paypal/android/card/CardAPIRequestFactory.kt

```diff
@@ -2,26 +2,37 @@ package com.paypal.android.card
 
 import com.paypal.android.core.APIRequest
 import com.paypal.android.core.HttpMethod
+import org.json.JSONObject
 
 internal class CardAPIRequestFactory {
 
     fun createConfirmPaymentSourceRequest(orderID: String, card: Card): APIRequest {
-        val path = "v2/checkout/orders/$orderID/confirm-payment-source"
-
         val cardNumber = card.number.replace("\\s".toRegex(), "")
         val cardExpiry = "${card.expirationYear}-${card.expirationMonth}"
 
-        val body = """
-            {
-                "payment_source": {
-                    "card": {
-                        "number": "$cardNumber",
-                        "expiry": "$cardExpiry",
-                        "security_code": "${card.securityCode}"
-                    }
-                }
-            }
-        """.trimIndent()
+        val cardJSON = JSONObject()
+            .put("number", cardNumber)
+            .put("expiry", cardExpiry)
+
+        card.cardholderName?.let { cardJSON.put("name", it) }
+        card.securityCode?.let { cardJSON.put("security_code", it) }
+
+        card.billingAddress?.let { billingAddress ->
```

Comment:
Instead of using the `let` scope function here, `apply` could be used. This way we'd have `this` as the object reference in the lambda, allowing us to clean up `billingAddress` from each of the calls to `.put()`.